pkg/controller/node/nodecontroller.go

```diff
@@ -532,6 +532,32 @@ func (nc *NodeController) tryUpdateNodeStatus(node *api.Node) (time.Duration, ap
 				readyCondition.LastTransitionTime = nc.now()
 			}
 		}
+
+		// Like NodeReady condition, NodeOutOfDisk was last set longer ago than gracePeriod, so update
+		// it to Unknown (regardless of its current value) in the master, without contacting kubelet.
+		oodCondition := nc.getCondition(&node.Status, api.NodeOutOfDisk)
+		if oodCondition == nil {
+			glog.V(2).Infof("Out of disk condition of node %v is never updated by kubelet", node.Name)
+			oodCondition := &api.NodeCondition{
```

Comment:
IMO, we should refactor and move all node status code out of kubelet.go. This should include factor out common code for setting conditions. That deserves its own PR though.